src/java.base/share/classes/sun/security/provider/NamedKEM.java

```diff
@@ -0,0 +1,218 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.security.provider;
+
+import sun.security.pkcs.NamedPKCS8Key;
+import sun.security.x509.NamedX509Key;
+
+import javax.crypto.DecapsulateException;
+import javax.crypto.KEM;
+import javax.crypto.KEMSpi;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.PrivateKey;
+import java.security.ProviderException;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.NamedParameterSpec;
+import java.util.Arrays;
+import java.util.Objects;
+
+/// A base class for all `KEM` implementations that can be
+/// configured with a named parameter set. See [NamedKeyPairGenerator]
+/// for more details.
+public abstract class NamedKEM implements KEMSpi {
+
+    private final String fname; // family name
+    private final String[] pnames; // allowed parameter set name (at least one)
+
+    /// Creates a new `NamedKEM` object.
+    ///
+    /// @param fname the family name
+    /// @param pnames the standard parameter set names, at least one is needed.
+    protected NamedKEM(String fname, String... pnames) {
+        this.fname = Objects.requireNonNull(fname);
+        if (pnames == null || pnames.length == 0) {
+            throw new AssertionError("pnames cannot be null or empty");
+        }
+        this.pnames = pnames;
+    }
+
+    @Override
+    public EncapsulatorSpi engineNewEncapsulator(PublicKey publicKey,
+            AlgorithmParameterSpec spec, SecureRandom secureRandom)
+            throws InvalidAlgorithmParameterException, InvalidKeyException {
+        if (spec != null) {
+            throw new InvalidAlgorithmParameterException("No params needed");
```

Comment:
I think we can just say "ML-KEM does not take any parameters". Otherwise it gives out a false hint that ML-KEM-768 does not because it already has a parameter set. The parameters in this method is not the parameters for the key but for the KEM algorithm itself. This is also the same for `NamedSignature::engineSetParameter`. It's even important there because one day when we decide to support PreHash ML-DSA we will need parameters.

Reply:
Ok, makes sense.